core/src/main/java/org/apache/iceberg/actions/BaseExpireSnapshotsActionResult.java

```diff
@@ -46,6 +48,22 @@ public BaseExpireSnapshotsActionResult(
     this.deletedEqDeleteFilesCount = deletedEqDeleteFilesCount;
     this.deletedManifestsCount = deletedManifestsCount;
     this.deletedManifestListsCount = deletedManifestListsCount;
+    this.deletedStatisticsFilesCount = 0;
+  }
+
+  public BaseExpireSnapshotsActionResult(
```

Comment:
Given that we have to continue creating new constructors, should we deprecate the old ones and no longer make the new ones public? If there is an interface, then there's no need for this class or its constructors to be public.

Reply:
I understood the deprecation part. Can you please elaborate on 

> no longer make the new ones public? If there is an interface, then there's no need for this class or its constructors to be public.

`BaseExpireSnapshotsActionResult` is used outside the package (in iceberg-spark). So, I still think it needs to be public.